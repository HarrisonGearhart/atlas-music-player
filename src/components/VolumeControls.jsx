import React from "react";

export default function VolumeControls() {
  return (
    <div className="flex items-center gap-2 w-48">
      <input
        type="range"
        min="0"
        max="100"
        className="w-full accent-emerald dark:accent-amethyst transition-colors duration-300"
      />
    </div>
  );
}
