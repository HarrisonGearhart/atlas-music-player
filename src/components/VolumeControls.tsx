import React from "react";

type VolumeControlsProps = {
  volume: number;
  setVolume: (value: number) => void;
  playbackRate: number;
  setPlaybackRate: (value: number) => void;
};

export default function VolumeControls({
  volume,
  setVolume,
  playbackRate,
  setPlaybackRate,
}: VolumeControlsProps) {
  return (
    <div className="flex flex-col gap-4">
      {/* Volume Slider */}
      <div className="flex items-center gap-2">
        <label className="text-sm text-[var(--color-text)]">Volume:</label>
        <input
          type="range"
          min={0}
          max={100}
          value={volume}
          onChange={(e) => setVolume(Number(e.target.value))}
          className="w-full"
        />
        <span className="text-sm">{volume}%</span>
      </div>

      {/* Playback Rate */}
      <div className="flex items-center gap-2">
        <label className="text-sm text-[var(--color-text)]">Speed:</label>
        <select
          value={playbackRate}
          onChange={(e) => setPlaybackRate(Number(e.target.value))}
          className="px-2 py-1 rounded border border-gray-400 bg-[var(--color-midnight)] text-[var(--color-text)]"
        >
          {[0.5, 0.75, 1, 1.25, 1.5, 2].map((rate) => (
            <option key={rate} value={rate}>
              {rate}x
            </option>
          ))}
        </select>
      </div>
    </div>
  );
}
