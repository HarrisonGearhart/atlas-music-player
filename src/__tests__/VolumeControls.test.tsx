import React from "react";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import VolumeControls from "../components/VolumeControls";

describe("VolumeControls Component", () => {
  it("renders with default volume 50", () => {
    const { container } = render(<VolumeControls volume={50} setVolume={() => {}} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with volume 0", () => {
    const { container } = render(<VolumeControls volume={0} setVolume={() => {}} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with volume 100", () => {
    const { container } = render(<VolumeControls volume={100} setVolume={() => {}} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
