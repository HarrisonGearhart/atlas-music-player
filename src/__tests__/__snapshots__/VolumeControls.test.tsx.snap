// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls Component > renders with default volume 50 1`] = `
<div
  class="flex flex-col items-start text-left"
>
  <h2
    class="text-lg font-bold text-[var(--color-azure)] -ml-1"
  >
    Song Title
  </h2>
  <p
    class="text-sm text-[var(--color-amethyst)] -ml-1"
  >
    Artist
  </p>
</div>
`;

exports[`VolumeControls Component > renders with volume 0 1`] = `
<div
  class="flex flex-col items-start text-left"
>
  <h2
    class="text-lg font-bold text-[var(--color-azure)] -ml-1"
  >
    Song Title
  </h2>
  <p
    class="text-sm text-[var(--color-amethyst)] -ml-1"
  >
    Artist
  </p>
</div>
`;

exports[`VolumeControls Component > renders with volume 100 1`] = `
<div
  class="flex flex-col items-start text-left"
>
  <h2
    class="text-lg font-bold text-[var(--color-azure)] -ml-1"
  >
    Song Title
  </h2>
  <p
    class="text-sm text-[var(--color-amethyst)] -ml-1"
  >
    Artist
  </p>
</div>
`;
